tests/spin/test_lattice.py

```diff
@@ -539,3 +541,74 @@ def test_shape_error():
     lattice = "Octagon"
     with pytest.raises(ValueError, match="Lattice shape, 'Octagon' is not supported."):
         _generate_lattice(lattice=lattice, n_cells=n_cells)
+
+
+def test_neighbour_order_error():
+    r"""Test that an error is raised if neighbour order is greater than 1 when custom_edges are provided."""
+
+    vectors = [[0, 1], [1, 0]]
+    n_cells = [3, 3]
+    custom_edges = [[(0, 1)], [(0, 5)], [(0, 4)]]
+    with pytest.raises(
+        ValueError, match="custom_edges and neighbour_order cannot be specified at the same time"
+    ):
+        Lattice(n_cells=n_cells, vectors=vectors, neighbour_order=2, custom_edges=custom_edges)
```

Comment:
Technically you wouldn't get an error if you specified `neighbour_order=1` but the error message suggests so. I think the "best practice" would be to make neighbour_order accept `None` or use some sentinel value, and basically write `if neighbour_order is None: neighbour_order = 1` inside your function. And you can move the check up for neighbour_order and custom_edges up (before that check) so that this would raise an error that is more intuitive.

Reply:
I think this might add more confusion. This makes it so that the default value is None. We can probably remove this  error, and add a warning and make it so that custom_edges supersedes the neighbour_order? or please let me know if you have other ideas.

Reply:
No I think an error is better. In this case I think it's not that important, we can just leave it as is and simply add a note to the documentation about `neighbour_order` and `custom_edges`. Best we edit the error message to be more specific like "Neighbour order cannot be greater than 1 when custom_edges is defined." though (IMO).

(Not important, read if you would like to learn about python i guess)
The alternative would be to use a sentinel value e.g. `DEFAULT = object()` and do `neighbour_order=DEFAULT` for the function signature. By doing this, you allow the check for `if neighbour_order is DEFAULT: neighbour_order = 1`. This avoids allowing a user to set `neighbour_order` equal to `None` e.g. `Lattice(...neighbour_order=None)` would not be valid as it would be if we let neighbour_order default to `None`.

This is the pythonic way of doing things but it is also... a little cumbersome imo especially since it's likely not to be a very important thing.